### PR TITLE
[fix] safely join url segments in manifest

### PR DESCRIPTION
**Pull request by benmccann**
fixes https://github.com/sveltejs/kit/issues/7641

### CODE REVIEW

File: packages/kit/src/core/generate_manifest/index.js [hunk 1]
```diff
@@ -111,14 +111,32 @@ export function generate_manifest({ build_data, relative_path, routes, format =
...
 		}
 	}`.replace(/^\t/gm, '');
 }
+
+/**
+ * @param {string} a
+ * @param {string} b
+ * @returns
+ */
+export function joinUrlSegments(a, b) {
```
Comment (Conduitry) [inline packages/kit/src/core/generate_manifest/index.js:131]: This is an internal API, so it should use snake case. I also don't think it needs to be exported from this module.

Comment (benmccann) [reply packages/kit/src/core/generate_manifest/index.js]: yep, thanks. I swiped this from Vite and forgot to update it :smile:

**Comment by Rich-Harris** [comment]
Can't we just change `relativePath: './'` to `relativePath: '.'`? I'm not a fan of loosey goosey functions that let you be careless about inputs

**Comment by Rich-Harris** [comment]
https://github.com/sveltejs/kit/pull/7655

**Comment by benmccann** [comment]
I tend to think along the lines of be liberal in what you accept and conservative in what you send: https://en.wikipedia.org/wiki/Robustness_principle

That being said, either one works, so I don't feel strongly

**Review by dummdidumm — APPROVED** [review]

**Pull request merged**